kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ProcessServiceIntegrationTest.java

```diff
@@ -589,6 +689,48 @@ public void testWorkItemOperationComplete() throws Exception {
             changeUser(TestConfig.getUsername());
         }
     }
+    
+    @Test
+    public void testWorkItemOperationCompleteWrongContainer() throws Exception {
+
+        Map<String, Object> parameters = new HashMap<String, Object>();
+
+        Long processInstanceId = processClient.startProcess(CONTAINER_ID, PROCESS_ID_EVALUATION, parameters);
+        try {
+            assertNotNull(processInstanceId);
+            assertTrue(processInstanceId.longValue() > 0);
+
+            // Completing human task so we can move in process flow to work item.
+            // User task shouldn't be handled as work item because in such case it doesn't behave consistently:
+            // i.e. leaving open tasks after finishing process instance.
+            List<String> status = Arrays.asList(Status.Ready.toString());
+            List<TaskSummary> taskList = taskClient.findTasksByStatusByProcessInstanceId(processInstanceId, status, 0, 10);
+
+            assertEquals(1, taskList.size());
+            TaskSummary taskSummary = taskList.get(0);
+            taskClient.startTask(CONTAINER_ID, taskSummary.getId(), USER_YODA);
+            taskClient.completeTask(CONTAINER_ID, taskSummary.getId(), USER_YODA, null);
+
+            TaskInstance userTask = taskClient.findTaskById(taskSummary.getId());
+            assertNotNull(userTask);
+            assertEquals("Evaluate items?", userTask.getName());
+            assertEquals(Status.Completed.toString(), userTask.getStatus());
+
+            List<WorkItemInstance> workItems = processClient.getWorkItemByProcessInstance(CONTAINER_ID, processInstanceId);
+            assertNotNull(workItems);
+            assertEquals(1, workItems.size());
+
+            WorkItemInstance workItemInstance = workItems.get(0);
+            assertNotNull(workItemInstance);
+
+            exceptionRule.expect(KieServicesException.class);
+            processClient.completeWorkItem(FUNNY_CONTAINER_ID, processInstanceId, workItemInstance.getId(), parameters);
+        } finally {
+            processClient.abortProcessInstance(CONTAINER_ID, processInstanceId);
+            changeUser(TestConfig.getUsername());
```

Comment:
I don't think this is needed - what do you think?

Reply:
I think we need to abort process instance always (probably it should a clean task for every test, but that will be a bigger change and will imply reviweing all test cases)

Reply:
mmm I was talking about the `changeUser` method call - I believe it's not needed here

Reply:
I think that is needed (it was already present in existing code) to leave the user as it was before the execution of the test